File: src/main/java/com/project/webapp/drivingschool/data/utils/LectureSeriesStatus.java
package com.project.webapp.drivingschool.data.utils;

/**
 * Status przebiegu cyklu wykładów
 */
public enum LectureSeriesStatus {

    /**
     * Status nowy
     * (jako jedyny daje możliwość dołączenia)
     */
    NEW,

    /**
     * Status trwający
     */
    ONGOING,

    /**
     * Status ukończony
     */
    FINISHED,

    /**
     * Status odwołany
     */
    CANCELED

}

File: frontend/src/app/utils/lecture-series-status.ts
/**
 * Status przebiegu cyklu wykładów.
 */
export enum LectureSeriesStatus {

  NEW,
  ONGOING,
  FINISHED

}

export namespace LectureSeriesStatus {

  const NEW_TRANSLATION = "Nowy";
  const ONGOING_TRANSLATION = "Trwający";
  const FINISHED_TRANSLATION = "Zakończony";

  export function translate(status: LectureSeriesStatus): string {
    let translator: string;
    switch (LectureSeriesStatus[status] as unknown) {
      case LectureSeriesStatus.NEW:
        translator = NEW_TRANSLATION;
        break;
      case LectureSeriesStatus.ONGOING:
        translator = ONGOING_TRANSLATION;
        break;
      case LectureSeriesStatus.FINISHED:
        translator = FINISHED_TRANSLATION;
        break;
      default:
        translator = "";
        break;
    }
    return translator;
  }

}
